### ISSUE TITLE
SkyblockBuilder is currently setup for only one island per world.

### ISSUE THREAD

**Issue by Ianthasys** [trulyno/star-technology#139]
It would be nice if the pack had setup skyblock builder in such a way that players spawn in on an spawn island, and then have the ability to use the built-in teams functionality to generate their own islands. This would allow players that play on servers to easily generate new islands for new players, or bring those new players into an existing team to join that player's progression on the pack. 

As of current, the mod is setup in such a way that the initial island that is spawned is the first player's island, and then all other players get spawned in there, as that's the world spawn. In order to make a new island for other players, one would have to bridge out a ways and build a new island and set the other player's spawn manually.

[Here's the link to the wiki for the mod](https://wiki.chaotictrials.de/skyblock-builder/1.20.x/packdev/packdev/) for further consideration

**Comment by trulyno** [comment]
I've skimmed over the wiki, but I could find any way to setup the world in the way you proposed. As far as I know, Skyblock builder works only by generating a central island from which players run a command to create their own island.
Adding a custom central island will not be in line with how the 1.19.2 branch of the pack has island generation, since it uses a different mod that is not on 1.20.1. We don't use Skyblock builder on 1.19.2 because of an issue with it that corrupts all recipes which has not happened on 1.20.1.

**Comment by Ianthasys** [comment]
According to the wiki, you're able to setup island templates, and then you're able to set those templates as either the initial spawn island, or set as a template for the other players that run the command to generate their own island. It doesn't seem that the wiki actually has any examples of how specifically to setup the spawn island other than brief reference to it [on the Main Setup page](https://wiki.chaotictrials.de/skyblock-builder/1.20.x/packdev/packdev/#:~:text=json5%20where%20needed.-,Setting%20a%20default%20spawn%20island,-You%20can%20set).

Looking at the  `/config/skyblockbuilder/templates.json5` in the 1.20.1 branch, it does seem that there is an example for this on lines 9-17 commented out from. I've added what I'd think is an example setup for the spawn island would look like in that file. It'd likely have a template under `/config/skyblockbuilder/templates` corresponding to whatever template is specified there.

```json5
  "spawn": {
     "name": "WorldSpawn",
     "desc": "",
     "file": "WorldSpawn.nbt",
     "spawns": "default",
     "offset": [ 0, 0, 0 ],
     "surroundingMargin": 0,
     "surroundingBlocks": "default"
 }
```

-----

> Adding a custom central island will not be in line with how the 1.19.2 branch of the pack has island generation, since it uses a different mod that is not on 1.20.1. We don't use Skyblock builder on 1.19.2 because of an issue with it that corrupts all recipes which has not happened on 1.20.1.

On this, I'm curious about the reasoning for continuing to support 1.19.2 while also maintaining the port to 1.20.1. This seems to me like quite an increased amount of work required for the project as well as having to deal with disparities between different versions of mods maybe missing features in the older version, while also holding back potential new features in the 1.20.1 version of the pack due to trying to maintain parity with the 1.19.2 version.

**Comment by trulyno** [comment]
Templates are simply island variants players can chose. We won't add any of those because we want parity between versions. I could not find an option in Skyblock Builder that would automatically create a personal island without a main island.

As for why we continue supporting 1.19.2, it is for those who started playing in 1.19.2. Converting your world brings risks of world corruption. Not everyone is willing to risk it. The reason why the 1.20.1 version exists is because the 1.19.2 version of GTm was deprecated while still in a buggy state(I have talked to the devs about a bugfix update to port only the fixes, but they refused). This situation practically forced me into making the 1.20.1 branch.
So in short, 1.20.1 branch is the main branch, while 1.19.2 branch is the continued support for those who started in 1.19.2 and do not wish to update to 1.20.1(for example, most of the youtubers that have a series on the pack have decided to not update to 1.20.1).
Because I'm developing this for the community that formed(and not for myself), continuing support for 1.19.2 is the correct choice in my opinion.

**Comment by Ianthasys** [comment]
> Templates are simply island variants players can chose.

You're actually able to specify which templates are ones that players can choose from, meaning a spawn island with signs/information for players to create their own islands could be left out from those specified templates.

Either way, I can see that this would probably be more hassle for you and the team than not, so I'll close the issue. Thank you for elaborating on the reasoning for continuing to support 1.19.2 for me. Most of the other players I talk to are in gen chat on the discord, and the folk in there on 1.19.2 have been somewhat rare as of late. I'd wrongly assumed that like myself, more players had jumped ship to the new versions than not, and the clarification helped to rectify that assumption.
